qemu/tests/cfg/qmp_command.cfg

```diff
@@ -3,200 +3,181 @@
+            check_item_in_pair = "no"
+            strict_match = "no"
             post_cmd = dmesg | grep virtio_blk
             # 62914560 512-byte logical blocks
-            cmd_return_value = 62914560
+            cmd_return_value = "['new size: 62914560']"
             event_cmd_type = guest_cmd
         - qmp_negative:
             qmp_cmd = subcommands
-            negative_test = yes
+            check_item_in_pair = "no"
+            cmd_result_check = equal
+            cmd_return_value = "["unknown command: 'subcommands'\r\n"]"
         - qmp_human-monitor-command:
             qmp_cmd = human-monitor-command command-line=info kvm
-            cmd_result_check = contain
-            cmd_return_value = "kvm support: enabled"
+            cmd_result_check = equal
+            check_item_in_pair = "no"
+            cmd_return_value = "['kvm support: enabled\r\n']"
         - qmp__com.redhat.drive_del:
-            only Host_RHEL.m6
+            only Host_RHEL.m6 Host_RHEL.m7
             pre_command += "qemu-img create -f qcow2 /tmp/__com.redhat.drive_del01.qcow2 100M"
-            event_cmd_type = qmp_cmd
             pre_cmd = "__com.redhat_drive_add file=/tmp/__com.redhat.drive_del01.qcow2, format=qcow2, id=com.redhat.drive_del01;query-block"
             qmp_cmd = "__com.redhat_drive_del id=com.redhat.drive_del01"
             post_cmd = "query-block"
             cmd_result_check = post_not_contain
-            cmd_return_value = com.redhat.drive_del01
+            check_item_in_pair = "no"
+            cmd_return_value = "['com.redhat.drive_del01']"
             post_command += " rm -rf /tmp/__com.redhat.drive_del01.qcow2;"
         - qmp_netdev_del:
             nics += " del01"
             netdev_id_del01 = "netdev_del01"
             device_id_del01 = "device_del01"
             ifname_del01 = "ifname_del01"
-            qmp_cmd = "netdev_del id=netdev_del01"
-            pre_cmd = info network
-            post_cmd = info network
-            event_cmd_type = monitor_cmd
+            qmp_cmd = "device_del id=device_del01; netdev_del id=netdev_del01"
+            post_cmd = "human-monitor-command command-line=info network"
             cmd_result_check = post_not_contain
-            cmd_return_value = netdev_del01:
-        - qmp_query-balloon:
-            balloon = balloon0
-            balloon_dev_devid = balloon0
-            balloon_dev_add_bus = yes
-            event_cmd_type = monitor_cmd
-            qmp_cmd = "query-balloon"
-            post_cmd = "info balloon"
-            cmd_result_check = m_equal_q
+            check_item_in_pair = "no"
+            strict_match = "no"
+            cmd_return_value = "['device_del01', 'netdev_del01']"
         - qmp_query-pci:
             qmp_cmd = "query-pci"
             cmd_result_check = contain
-            cmd_return_value = "function;regions;qdev"
+            check_item_in_pair = "no"
+            cmd_return_value = "['function', 'regions', 'qdev_id']"
```

Comment:
Can we change the verified value?
As I understanding, it's not a effect method for checking it. should we check the device name like ""desc": "ISA bridge" "IDE controller" ...."
By the way, a little difference between pc and q35 machine type, "IDE controller" for pc and "SATA controller" for q35

Thanks
Jing

Reply:
@yanan-fu  no specific polarion test case for it. I can list the detailed info here
q35 result:
{"return": [{"bus": 0, "devices": [{"bus": 0, "qdev_id": "", "slot": 0, "class_info": {"class": 1536, "desc": "Host bridge"}, "id": {"device": 10688, "subsystem-vendor": 6900, "vendor": 32902, "subsystem": 4352}, "function": 0, "regions": []}, {"bus": 0, "qdev_id": "", "slot": 1, "class_info": {"class": 768, "desc": "VGA controller"}, "id": {"device": 4369, "subsystem-vendor": 6900, "vendor": 4660, "subsystem": 4352}, "function": 0, "regions": [{"prefetch": true, "mem_type_64": false, "bar": 0, "size": 16777216, "address": 4244635648, "type": "memory"}, {"prefetch": false, "mem_type_64": false, "bar": 2, "size": 4096, "address": 4273815552, "type": "memory"}, {"prefetch": false, "mem_type_64": false, "bar": 6, "size": 65536, "address": -1, "type": "memory"}]}, {"bus": 0, "qdev_id": "", "irq": 11, "slot": 2, "class_info": {"class": 512, "desc": "Ethernet controller"}, "id": {"device": 4307, "subsystem-vendor": 32902, "vendor": 32902, "subsystem": 0}, "function": 0, "regions": [{"prefetch": false, "mem_type_64": false, "bar": 0, "size": 131072, "address": 4273471488, "type": "memory"}, {"prefetch": false, "mem_type_64": false, "bar": 1, "size": 131072, "address": 4273602560, "type": "memory"}, {"bar": 2, "size": 32, "address": 49216, "type": "io"}, {"prefetch": false, "mem_type_64": false, "bar": 3, "size": 16384, "address": 4273799168, "type": "memory"}, {"prefetch": false, "mem_type_64": false, "bar": 6, "size": 262144, "address": -1, "type": "memory"}]}, {"bus": 0, "qdev_id": "", "slot": 31, "class_info": {"class": 1537, "desc": "ISA bridge"}, "id": {"device": 10520, "subsystem-vendor": 6900, "vendor": 32902, "subsystem": 4352}, "function": 0, "regions": []}, {"bus": 0, "qdev_id": "", "irq": 10, "slot": 31, "class_info": {"class": 262, "desc": "SATA controller"}, "id": {"device": 10530, "subsystem-vendor": 6900, "vendor": 32902, "subsystem": 4352}, "function": 2, "regions": [{"bar": 4, "size": 32, "address": 49248, "type": "io"}, {"prefetch": false, "mem_type_64": false, "bar": 5, "size": 4096, "address": 4273819648, "type": "memory"}]}, {"bus": 0, "qdev_id": "", "irq": 10, "slot": 31, "class_info": {"class": 3077, "desc": "SMBus"}, "id": {"device": 10544, "subsystem-vendor": 6900, "vendor": 32902, "subsystem": 4352}, "function": 3, "regions": [{"bar": 4, "size": 64, "address": 1792, "type": "io"}]}]}]}

pc result:
{"return": [{"bus": 0, "devices": [{"bus": 0, "qdev_id": "", "slot": 0, "class_info": {"class": 1536, "desc": "Host bridge"}, "id": {"device": 4663, "subsystem-vendor": 6900, "vendor": 32902, "subsystem": 4352}, "function": 0, "regions": []}, {"bus": 0, "qdev_id": "", "slot": 1, "class_info": {"class": 1537, "desc": "ISA bridge"}, "id": {"device": 28672, "subsystem-vendor": 6900, "vendor": 32902, "subsystem": 4352}, "function": 0, "regions": []}, {"bus": 0, "qdev_id": "", "slot": 1, "class_info": {"class": 257, "desc": "IDE controller"}, "id": {"device": 28688, "subsystem-vendor": 6900, "vendor": 32902, "subsystem": 4352}, "function": 1, "regions": [{"bar": 4, "size": 16, "address": 49216, "type": "io"}]}, {"bus": 0, "qdev_id": "", "irq": 9, "slot": 1, "class_info": {"class": 1664, "desc": "Bridge"}, "id": {"device": 28947, "subsystem-vendor": 6900, "vendor": 32902, "subsystem": 4352}, "function": 3, "regions": []}, {"bus": 0, "qdev_id": "", "slot": 2, "class_info": {"class": 768, "desc": "VGA controller"}, "id": {"device": 4369, "subsystem-vendor": 6900, "vendor": 4660, "subsystem": 4352}, "function": 0, "regions": [{"prefetch": true, "mem_type_64": false, "bar": 0, "size": 16777216, "address": 4244635648, "type": "memory"}, {"prefetch": false, "mem_type_64": false, "bar": 2, "size": 4096, "address": 4273930240, "type": "memory"}, {"prefetch": false, "mem_type_64": false, "bar": 6, "size": 65536, "address": -1, "type": "memory"}]}, {"bus": 0, "qdev_id": "", "irq": 11, "slot": 3, "class_info": {"class": 512, "desc": "Ethernet controller"}, "id": {"device": 4110, "subsystem-vendor": 6900, "vendor": 32902, "subsystem": 4352}, "function": 0, "regions": [{"prefetch": false, "mem_type_64": false, "bar": 0, "size": 131072, "address": 4273733632, "type": "memory"}, {"bar": 1, "size": 64, "address": 49152, "type": "io"}, {"prefetch": false, "mem_type_64": false, "bar": 6, "size": 262144, "address": -1, "type": "memory"}]}]}]}

Reply:
@yanan-fu 
could you help to add a check for "desc": VGA controller,Ethernet controller,ISA bridge and IDE controller/SATA controller?

Thanks
Jing

Reply:
@jingzhao84 
pc: 'Host bridge', 'ISA bridge', 'IDE controller'   and  q35: 'Host bridge', 'ISA bridge' 'SATA controller' is ok for me.

But, VGA controller and Ethernet controller don't exist with minimal qemu command line and '-nodefaults'.   Do we still need to check them?

Reply:
It's ok, we can ignore vga and Ethernet controller check

Reply:
@yanan-fu double confirm it and we should keep it since covered by other place